### PR TITLE
Prevent track contact just by embed form

### PR DESCRIPTION
**Pull request by kuzmany**
**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | Yes
| New feature? | 
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/6034
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
Based on this conversations https://github.com/mautic/mautic/issues/6034,  the pull request prevent create contact (https://github.com/kuzmany/mautic/blob/d039cb16e04aadecd996b733806577d0eefeecb8/app/bundles/FormBundle/Model/FormModel.php#L806)  until you create first field with Auto fill data option

#### Steps to reproduce bug
Steps to reproduce:
1. Create form and call mauticinstance/form/generate.js?id=xx
2. go on contacts > anonymous
3. see anonymouse  contact here with your IP

#### Steps to test this PR:
1.  Repeat all steps
2. See If anonymouse contact is not tracked until you create form field with enable Auto fill data

### CODE REVIEW

**Comment by kuzmany** [comment]
Label: WIP

File: app/bundles/FormBundle/Model/FormModel.php [hunk 1]
```diff
@@ -788,24 +788,31 @@ public function populateValuesWithGetParameters(Form $form, &$formHtml)
...
 
-        $fields = $form->getFields();
-        /** @var \Mautic\FormBundle\Entity\Field $f */
         foreach ($fields as $f) {
-            $leadField  = $f->getLeadField();
-            $isAutoFill = $f->getIsAutoFill();
-
-            if (isset($leadField) && $isAutoFill) {
-                $value = $lead->getFieldValue($leadField);
+            $value = $lead->getFieldValue($leadField);
```
Comment (alanhartless) [inline app/bundles/FormBundle/Model/FormModel.php]: Shouldn't this be `$f`? Although might be better to rename `$f` to something more descriptive and update in line 815 as well.

**Comment by npracht** [comment]
@alanhartless is that ok for you? Then we could fix it for 2.14 :)

File: app/bundles/FormBundle/Model/FormModel.php [hunk 2]
```diff
@@ -788,24 +788,31 @@ public function populateValuesWithGetParameters(Form $form, &$formHtml)
...
             return;
         }
 
-        $fields = $form->getFields();
-        /** @var \Mautic\FormBundle\Entity\Field $f */
-        foreach ($fields as $f) {
-            $leadField  = $f->getLeadField();
-            $isAutoFill = $f->getIsAutoFill();
+        foreach ($fields as $field) {
+            $value = $lead->getFieldValue($leadField);
```
Comment (alanhartless) [inline app/bundles/FormBundle/Model/FormModel.php]: This is still fetching the wrong field name. It's fetching the value of the field consumed in the foreach of 793. $leadField should be $field inside this loop.

Comment (kuzmany) [reply app/bundles/FormBundle/Model/FormModel.php]: Fixed

File: app/bundles/FormBundle/Model/FormModel.php (hunk 1 above)
Comment (npracht) [reply app/bundles/FormBundle/Model/FormModel.php]: That has been addressed also

**Comment by npracht** [comment]
@alanhartless i think everything is alright now from a code perspective based on your comments. If you agree that's `Ready to commit` !

**Comment by escopecz** [comment]
The same problem as https://github.com/mautic/mautic/pull/5400#issuecomment-412486935 is here.

@alanhartless are the requested changes OK now?

**Pull request merged**